From: Adam Jackson <ajax@redhat.com>
Date: Sun, 28 Oct 2007 09:37:52 +0100
Subject: [PATCH] Fix accidental ABI usage in RANDR 1.2 drivers.

Due to RANDR 1.2, xf86findOptionValue and xf86nameCompare are now ABI.
Make sure they're exported from the server.

---
diff -up xorg-server-1.3.0.0/hw/xfree86/parser/Flags.c.jx xorg-server-1.3.0.0/hw/xfree86/parser/Flags.c
--- xorg-server-1.3.0.0/hw/xfree86/parser/Flags.c.jx	2006-09-18 02:04:18.000000000 -0400
+++ xorg-server-1.3.0.0/hw/xfree86/parser/Flags.c	2007-10-03 11:08:01.000000000 -0400
@@ -330,7 +330,7 @@ xf86findOption (XF86OptionPtr list, cons
  * returned.  If the option is not found, a NULL is returned.
  */
 
-char *
+__attribute__((visibility("default"))) char *
 xf86findOptionValue (XF86OptionPtr list, const char *name)
 {
 	XF86OptionPtr p = xf86findOption (list, name);
diff -up xorg-server-1.3.0.0/hw/xfree86/parser/scan.c.jx xorg-server-1.3.0.0/hw/xfree86/parser/scan.c
--- xorg-server-1.3.0.0/hw/xfree86/parser/scan.c.jx	2006-10-04 17:46:03.000000000 -0400
+++ xorg-server-1.3.0.0/hw/xfree86/parser/scan.c	2007-10-03 10:57:43.000000000 -0400
@@ -985,7 +985,7 @@ StringToToken (char *str, xf86ConfigSymT
  * Compare two names.  The characters '_', ' ', and '\t' are ignored
  * in the comparison.
  */
-int
+__attribute__((visibility("default"))) int
 xf86nameCompare (const char *s1, const char *s2)
 {
 	char c1, c2;
